packages/api/src/external/hie/get-hie-initiator.ts

```diff
@@ -14,17 +15,61 @@ export type HieInitiator = {
 
 export async function getHieInitiator(
   patient: Pick<Patient, "id" | "cxId">,
+  facilityId: string | undefined
+): Promise<HieInitiator> {
+  const { organization, facilities } = await getPatientWithDependencies(patient);
+
+  const facility = await getPatientsFacility(patient.id, facilities, facilityId);
+
+  if (isItVendor(organization.type)) {
+    return {
+      oid: facility.oid,
+      name: facility.data.name,
+      npi: facility.data.npi,
+      facilityId: facility.id,
+      orgName: organization.data.name,
+    };
+  }
+  return {
+    oid: organization.oid,
+    name: organization.data.name,
+    npi: facility.data.npi,
+    facilityId: facility.id,
+    orgName: organization.data.name,
+  };
+}
+
+export async function isHieEnabledToQuery(
   facilityId: string | undefined,
-  // had to specify them instead of using the type because of the item ALL
+  patient: Pick<Patient, "id" | "cxId">,
   hie: MedicalDataSource.COMMONWELL | MedicalDataSource.CAREQUALITY
-): Promise<HieInitiator> {
+): Promise<boolean> {
   const { organization, facilities } = await getPatientWithDependencies(patient);
+
+  const facility = await getPatientsFacility(patient.id, facilities, facilityId);
+
+  if (isItVendor(organization.type)) {
+    const facilityType = hie === MedicalDataSource.COMMONWELL ? facility.cwType : facility.cqType;
+    if (isOboFacility(facilityType) && !isOboEnabledForHie(facility, hie)) {
```

Comment:
once we update the `enabled` flags, all we should do is check whether the facility is enabled